peps/pep-0733.rst

```diff
@@ -0,0 +1,627 @@
+the `capi-workgroup <https://github.com/capi-workgroup/problems/issues/>`__ repository.
+The issues are grouped into several categories.
+
+
+API Evolution and Maintenance
+-----------------------------
+
+The difficulty of making changes in the C API is central to this report. It is
+implicit in many of the issues we discuss here, particularly when we need to
+decide whether an incremental bugfix can resolve the issue, or whether it can
+only be addressed as part of an API redesign
+[`Issue 44 <https://github.com/capi-workgroup/problems/issues/44>`__]. The
+benefit of each incremental change is often viewed as too small to justify the
+disruption. Over time, this implies that every mistake we make in an API's
+design or implementation remains with us indefinitely.
+
+We can take two views on this issue. One is that this is a problem and the
+solution needs to be baked into any new C API we design, in the form of a
+process for incremental API evolution. The other possible approach is that
+this is not a problem to be solved, but rather a feature of any API. In this
+view, API evolution should not be incremental, but rather through large
+redesigns, each of which learns from the mistakes of the past and is not
+shackled by backwards compatibility requirements. A compromise approach
+is somewhere between these two extremes, fixing issues which are easy
+or important enough to tackle incrementally, and leaving others alone.
+
+The problem we have in CPython is that we don't have an agreed, official
+approach to API evolution. Different members of the core team are pulling in
+different directions and this is an ongoing source of disagreements.
+Any new C API needs to come with a clear decision about the model
+that its maintenance will follow, as well as the technical and
+organizational processes by which this will work.
+
+If the model does include provisions for incremental evolution of the API,
+it will include processes for managing the impact of the change on users
+[`Issue 60 <https://github.com/capi-workgroup/problems/issues/60>`__],
+perhaps through introducing an external backwards compatibility module
+[`Issue 62 <https://github.com/capi-workgroup/problems/issues/62>`__],
+or a new API tier of "blessed" functions
+[`Issue 55 <https://github.com/capi-workgroup/problems/issues/55>`__].
+
+
+API Specification and Abstraction
+---------------------------------
+
+The C API does not have a formal specification, it is described
+semi-formally in the documentation and exposed through C header
+files. This creates a number of problems.
+
+Bindings for languages other than C/C++ must parse C code
+[`Issue 7 <https://github.com/capi-workgroup/problems/issues/7>`__].
+Some C language features are hard to handle in this way, because
+they produce compiler-dependent output (such as enums) or require
+a C preprocessor/compiler rather than just a parser (such as macros)
+[`Issue 35 <https://github.com/capi-workgroup/problems/issues/35>`__].
```

Comment:
Things that can also be problematic for "less complete" C FFIs like .NET's are
- accessing non-function symbols (`PyNone` etc.)
- functions with variadic arguments

Reply:
> accessing non-function symbols (PyNone etc.)

Do you think that the PEP doesn't already explain this issue properly? Do you want to propose to rephrase or elaborate a section?

> functions with variadic arguments

see line 507

Reply:
I'll have another look, maybe I missed this.

Reply:
Ah, I see, I'll add non-function symbols to the list as a suggestion :)